source/PlayerInfo.cpp

```diff
@@ -841,15 +841,32 @@ void PlayerInfo::AddShip(const shared_ptr<Ship> &ship)
 
 
 
-// Buy a ship of the given model, and give it the given name.
-void PlayerInfo::BuyShip(const Ship *model, const string &name)
+// Charge the player the cost of a ship and give it the specified name:
+void PlayerInfo::BuyShip(const Ship *model, const std::string &name)
+{
+	ReceiveShip(model, name, DEFAULT_COST, true);
+}
+
+
+
+// Receive a ship as a gift
+shared_ptr<Ship> PlayerInfo::GiftShip(const Ship *model, const std::string &name, bool record_depriciation)
+{
+	return ReceiveShip(model, name, 0, record_depriciation);
+}
+
+
+
+// Purchase or receive a gift of a ship of the given model and name
+shared_ptr<Ship> PlayerInfo::ReceiveShip(const Ship *model, const string &name, int64_t cost, bool record_depriciation)
 {
 	if(!model)
-		return;
+		return nullptr;
 	
 	int day = date.DaysSinceEpoch();
-	int64_t cost = stockDepreciation.Value(*model, day);
-	if(accounts.Credits() >= cost)
+	if(cost == DEFAULT_COST)
```

Comment:
This DEFAULT_COST seems clumsy to me. The cost parameter can ONLY ever be one of two things; either it is 0, or it is DEFAULT_COST. We should just be using a boolean. This same boolean can replace recordDepreciation, as the cost is always tied to that boolean right now; if cost is 0, then we know that recordDepreciation is false. If cost is DEFAULT_COST, then we know recordDepreciation is true. So just create a single `gifting` bool that is true for GiftShip and false for BuyShip.

Reply:
Yeah, that makes sense. Made the changes